domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -461,7 +507,19 @@ private void init() throws Exception {
       startEmbeddedPostgres();
     }
 
-    client = io.vertx.ext.asyncsql.PostgreSQLClient.createNonShared(vertx, postgreSQLClientConfig);
+    client = null;
+    startPool();
+
+    log.debug("Setup timer = " + activityTimerId + " for tenant " + tenantId);
+    activityTimerId = vertx.setPeriodic(postgreSQLClientConfig.getInteger(CONNECTION_RELEASE_DELAY), res -> {
+      log.debug("expire timer for tenant " + tenantId);
+      if (!pgPoolActivity && client != null) {
+        log.debug("Expire pool for tenant " + tenantId);
+        client.close();
+        client = null;
+      }
+      pgPoolActivity = false;
+    });
```

Comment:
The expire timer should be split into a separate pull request so that we can discuss it separately.

Reply:
A module can create any number of PostgresClients, if a PostgresClient is no longer used does the timer stop and allow the garbage collection to free the PostgresClient?
What happens if a bulk insert or a transaction takes three minutes? Does the expire timer cancel it?

Reply:
I don't think it should be a separate PR. We can't have a situation where connections are never released.

For same tenant only one PostgresClient is returned (it's a static constructor).

Connections that are returned by pool are not released. Only the ones that are unused.· So w could release every second if we wanted to.

Reply:
ok. Since it does not work as I thought it does, I'll take all this out. This means we probably can't use vertx-pg-client as it stands.